src/chi-vue/src/components/column-customization/ColumnCustomization.tsx

```diff
@@ -96,20 +97,47 @@ export default class ColumnCustomization extends Vue {
     if (this._ColumnCustomizationContentComponent) {
       this._availableColumns = [];
       this._selectedColumns = [];
+      this._selectedData = this.columnsData?.columns.filter((column: DataTableColumn) => column.selected);
       this._processData();
-      (this.$refs.saveButton as HTMLButtonElement).disabled = true;
+      (this.$refs.saveButton as HTMLButtonElement).disabled = false;
       (this.$refs.resetButton as HTMLButtonElement).disabled = true;
       this.key += 1;
     }
-    this.$emit(DATA_TABLE_EVENTS.COLUMNS_RESET, [...(this._selectedColumns || [])]);
   }
 
   _submitColumnsChange() {
+    this._previousSelected = this._selectedData;
     this.$emit(DATA_TABLE_EVENTS.COLUMNS_CHANGE, this._selectedData);
     (this.$refs.saveButton as HTMLButtonElement).disabled = true;
     this._chiModal.hide();
   }
 
+  _cancelColumnsChange() {
+    const originalSelectedColumns = this.columnsData?.columns.filter((column: DataTableColumn) => column.selected);
+
+    if (this._previousSelected) {
+      this._selectedData = this._previousSelected;
```

Comment:
Cancel button was supposed to close modal only. Are we changing the behavior?

Reply:
Yes we are, now the cancel button returns the previous state of the selected and availables columns, and closing the modal.